packages/admin/admin/src/dataGrid/CrudContextMenu.tsx

```diff
@@ -57,14 +68,25 @@ export function CrudContextMenu<CopyData>({ url, onPaste, onDelete, refetchQueri
     const [deleteDialogOpen, setDeleteDialogOpen] = React.useState(false);
     const [copyLoading, setCopyLoading] = React.useState(false);
     const [pasting, setPasting] = React.useState(false);
+    const [deleteLoading, setDeleteLoading] = React.useState(false);
+    const [hasDeleteErrors, setHasDeleteErrors] = React.useState(false);
 
     const handleDeleteClick = async () => {
         if (!onDelete) return;
-        await onDelete({
-            client,
-        });
-        if (refetchQueries) await client.refetchQueries({ include: refetchQueries });
-        setDeleteDialogOpen(false);
+        setHasDeleteErrors(false);
+        setDeleteLoading(true);
+        try {
+            await onDelete({
+                client,
+            });
+            if (refetchQueries) await client.refetchQueries({ include: refetchQueries });
+            setDeleteDialogOpen(false);
```

Comment:
Shouldn't this be in the finally-branch?

Reply:
No because you don't want the dialog to close on error :)